glyphs-reader/src/font.rs

```diff
@@ -550,20 +590,45 @@ impl RawFont {
         Ok(())
     }
 
-    fn v2_to_v3_shapes(&mut self) -> Result<(), Error> {
+    fn v3_shapes_to_v2_paths_and_components(&mut self) -> Result<(), Error> {
         // shapes in v3 encompasses paths and components in v2
+        // paths and components is closer to what IR wants
+        // so here, unusually, downgrade
         for glyph in self.glyphs.iter_mut() {
             // Paths and components combine in shapes
-            // TODO components
             for layer in glyph.layers.iter_mut() {
-                if let Some(paths) = layer.paths.take() {
-                    match &mut layer.shapes {
-                        Some(shapes) => {
-                            shapes.extend(paths);
-                        }
-                        None => {
-                            layer.shapes = Some(paths);
-                        }
+                let Some(mut shapes) = layer.shapes.take() else {
+                    continue;
+                };
+
+                for shape in shapes.iter_mut() {
+                    if let Some(Plist::String(ref_glyph_name)) = shape.other_stuff.remove("ref") {
+                        // only components use ref
+                        let transform = if let Some(pos) = &shape.pos {
+                            if pos.len() != 2 {
+                                return Err(Error::StructuralError(format!(
+                                    "Strang pos {:?}",
+                                    pos
+                                )));
+                            }
+                            Some(Affine::translate(pos[0].into_inner(), pos[1].into_inner()))
+                        } else {
+                            None
```

Comment:
I would make Component::transform just Affine instead of Option<Affine>, and set to identity by default

Reply:
I believe it has to be Option due to plist parsing

Reply:
Done